library/pk.c

```diff
@@ -329,12 +329,7 @@ int mbedtls_pk_check_pair( const mbedtls_pk_context *pub, const mbedtls_pk_conte
         return( MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE );
     }
 
-    if( prv->pk_info->type == MBEDTLS_PK_RSA_ALT )
```

Comment:
Blocker: removing this here is incorrect: the code below is going to return `PK_TYPE_MISMATCH` before `rsa_alt_check_pair` is called.

Reply:
@AndrzejKurek Please add a non-regression test for this.

Reply:
This will be added as https://jira.arm.com/browse/IOTSSL-2066. Right now we have to ship this PR as soon as possible, since Azim's work depends on it.